### PR TITLE
Versioning -feature

### PR DESCRIPTION
**Pull request by sorja**
Adds db versioning feature for admins

### CODE REVIEW

File: webapp/loggedin/admin/components/versioningView/versioningView.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    </Switch>
+  )
+}
+
+VersioningView.defaultProps = {
+  versions: []
+}
+
+const mapStateToProps = (state) => ({
+  versions: state.admin.versions
```
Comment (minotogna) [inline webapp/loggedin/admin/components/versioningView/versioningView.js]: should we start creating state objects?

File: server/system/schedulers/versioningScheduler.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      return
+    }
+
+    versions.forEach(async (version) => {
+      console.log('New version pending creation:')
+      console.table(version)
+
+      handleNewVersion(version)
+
+    });
```
[the remark this answers is not in the document]
Comment (sorja) [reply server/system/schedulers/versioningScheduler.js]: Thanks, forgot about forEach and async. Using for .. of .. instead

File: server/versioning/api.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+const { getVersionById, getAllVersions, addVersion, deleteVersion, getSchemaByName } = require('./versioningRepository')
+const { sendErr } = require('../utils/requestUtils')
+
+// TODO: add authentication
```
Comment (minotogna) [inline server/versioning/api.js]: why this TODO?

File: server/versioning/api.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+  app.post('/versioning/', async (req, res) => {
+    const userId = req.user.id
+    const { version, timestamp } = req.body
+    try {
+      if (!version || !timestamp) {
+        console.log({
+          version,
+          timestamp
+        })
```
Comment (minotogna) [inline server/versioning/api.js]: I don't really like console.log in the code...

File: webapp/loggedin/admin/actions.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  // Major.Minor.Patch
+  // <num>.<num>.<num> ex. 1.0.0
+  const versionValid = /\d+\.\d+\.\d+/.test(newVersionForm.version)
+  if (!versionValid) {
+    dispatch({
+      type: versioningPostMissingData
+    })
+    return
+  }
+
```
Comment (minotogna) [inline webapp/loggedin/admin/actions.js]: do we check that versions are consecutive?

File: webapp/loggedin/admin/components/versioningView/NewVersionForm.js [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+//           addNewUser(countryIso)
+//           setAdding(false)
+//         }}>
+//           {i18n.t('userManagement.addUser')}
+//         </button>
+//       </td>
+//     </tr>
+//   </tbody>
+// </table>
+// </div>
```
Comment (minotogna) [inline webapp/loggedin/admin/components/versioningView/NewVersionForm.js]: can we remove the commented code?

File: webapp/loggedin/admin/components/versioningView/versioningView.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    <Switch>
+      <Route exact path={path}>
+        {versionsExist ?
+          <VersioningViewTable
+            i18n={i18n}
+            deleteVersion={deleteVersion}
+            versions={versions}
+            getVersions={getVersions} />
+          :
+          <h1>No versions yet</h1>}
```
Comment (minotogna) [inline webapp/loggedin/admin/components/versioningView/versioningView.js]: we need to add this to i18n as well

File: server/versioning/api.js (hunk 3 above)
Comment (sorja) [reply server/versioning/api.js]: Updated, user must be admin

File: server/versioning/api.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
+const { getAllVersions, addVersion, deleteVersion } = require('./versioningRepository')
+const { isAdministrator } = require("../../common/countryRole")
+const { sendErr, getUser, send404 } = require('../utils/requestUtils')
+
+// TODO: add authentication
```
Comment (minotogna) [inline server/versioning/api.js]: maybe we can remove the TODO?

File: server/versioning/api.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+module.exports.init = app => {
+  app.get('/versioning/', async (req, res) => {
+    const user = getUser(req)
+    if (!isAdministrator(user)) {
+      send404(res)
+    }
```
Comment (minotogna) [inline server/versioning/api.js]: this code is duplicated. what about having 
```
const checkAdmin = (req, res) => {
    const user = getUser(req)
      if (!isAdministrator(user)) {
        send404(res)
      }
}
```

**Pull request merged**